File: androidSharedNative/src/main/java/io/github/lumkit/tweak/sharednative/NativeFileBridge.java
package io.github.lumkit.tweak.sharednative;

final class NativeFileBridge {

    static {
        System.loadLibrary("tweak_shared_native");
    }

    private NativeFileBridge() {
    }

    static native boolean exists(String path);

    static native byte[] readBytes(String path);

    static native void writeBytes(String path, byte[] bytes);

    static native void delete(String path, boolean recursive);

    static native String[] list(String path);

    static native void mkdirs(String path);

    static native void copy(String sourcePath, String targetPath, boolean overwrite);

    static native void move(String sourcePath, String targetPath, boolean overwrite);

    static native void chmod(String path, String mode);
}

File: androidSharedNative/src/main/kotlin/io/github/lumkit/tweak/sharednative/NativeFileBridge.kt
package io.github.lumkit.tweak.sharednative

internal object NativeFileBridge {

    init {
        System.loadLibrary("tweak_shared_native")
    }

    @JvmStatic
    external fun exists(path: String): Boolean

    @JvmStatic
    external fun readBytes(path: String): ByteArray

    @JvmStatic
    external fun writeBytes(path: String, bytes: ByteArray)

    @JvmStatic
    external fun delete(path: String, recursive: Boolean)

    @JvmStatic
    external fun list(path: String): Array<String>?

    @JvmStatic
    external fun mkdirs(path: String)

    @JvmStatic
    external fun copy(sourcePath: String, targetPath: String, overwrite: Boolean)

    @JvmStatic
    external fun move(sourcePath: String, targetPath: String, overwrite: Boolean)

    @JvmStatic
    external fun chmod(path: String, mode: String)

    @JvmStatic
    external fun length(path: String): Long
}
